### PR TITLE
adding a seed for np.random for deterministic failures

### LINKED ISSUE #124: Randomization in tests
Right now all of our tests are using random data. This sometimes causes github actions to fail in unexpected ways.

We should use all deterministic data in our tests rather than random so that we get consistency across runs.

### LINKED ISSUE #129: TestSklearnGradientBoostingConverter fails occasionally (~3% of time)
Now that we almost have our (win/mac/linux)x(3.5/3.6/3.7/3.8) pipeline done which calls the tests 12 times, it seems that `TestSklearnGradientBoostingConverter` fails about 1/36 runs. There doesn't seem to be a pattern as to which OS/version, so maybe this is related to the random data?  (see also #124 )

[Example](https://github.com/microsoft/hummingbird/runs/766149709?check_suite_focus=true):

```
y = array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int64)
sample_weight = array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.], dtype=float32)

    def _validate_y(self, y, sample_weight):
        check_classification_targets(y)
        self.classes_, y = np.unique(y, return_inverse=True)
        n_trim_classes = np.count_nonzero(np.bincount(y, sample_weight))
        if n_trim_classes < 2:
            raise ValueError("y contains %d class after sample_weight "
                             "trimmed classes with zero weights, while a "
                             "minimum of 2 classes are required."
>                            % n_trim_classes)
E           ValueError: y contains 1 class after sample_weight trimmed classes with zero weights, while a minimum of 2 classes are required.

c:\hostedtoolcache\windows\python\3.5.4\x64\lib\site-packages\sklearn\ensemble\_gb.py:2097: ValueError
=========================== short test summary info ===========================
FAILED tests/test_sklearn_gbdt_converters.py::TestSklearnGradientBoostingConverter::test_gbdt_implementation
======================== 1 failed, 86 passed in 36.85s ========================
##[error]Process completed with exit code 1.
 Coverage
```

### PR DESCRIPTION
**Pull request by ksaur**
Addressing Issue #124 and #129

### CODE REVIEW

File: tests/test_sklearn_histgbdt_converters.py [hunk 1]
```diff
@@ -17,9 +17,10 @@ def _run_GB_trees_classifier_converter(self, num_classes, extra_config={}, label
         warnings.filterwarnings("ignore")
         for max_depth in [2, 3, 8, 10, 12, None]:
             model = HistGradientBoostingClassifier(max_iter=10, max_depth=max_depth)
-            X = [[0, 1], [1, 1], [2, 0]]
+            np.random.seed(0)
```
Comment (interesaaat) [inline tests/test_sklearn_histgbdt_converters.py:20]: I fixed the data because `hist_gbdt` was generating errors quite often.

Comment (ksaur) [reply tests/test_sklearn_histgbdt_converters.py:20]: was this because of a SKL problem? or something else?

Comment (interesaaat) [reply tests/test_sklearn_histgbdt_converters.py:20]: It is something related to `hist_gbdt` not sklearn per me.

**Comment by ksaur** [comment]
This also increases coverage in GBDT: https://codecov.io/gh/microsoft/hummingbird/pull/130/changes

**Pull request merged**